keps/sig-node/2570-memory-qos/README.md

```diff
@@ -104,6 +107,145 @@ Brief map as follows:
+
+2. It could result in early throttling even when a sizeable chunk of memory is unused. 
+  
+    For example, 
+    * `requests.memory` = 800Mi
+
+      `memory throttling factor` = 0.8
+
+      `limits.memory` = 1000Mi
+
+      As per Alpha implementation,
+
+      `memory.high` = memory throttling factor * limits.memory  = 0.8 * 1000Mi = 800Mi
+
+      Throttling occurs at 800Mi, while a sizable chunk of 200Mi is still unused.
+
+    * `requests.memory` = 500Mi
+
+      `memory throttling factor` = 0.6
+
+      `limits.memory` = 1000Mi
+      
+      As per Alpha implementation,
+
+      `memory.high` = memory throttling factor * limits.memory = 0.6 * 1000Mi = 600Mi
+      
+      Throttling occurs at 600Mi which is just a 100Mi over the requested memory. 400Mi of memory is still unused as early throttling happens.
+  
+
+3. Default throttling factor of 0.8 might be too aggressive for some applications that are latency sensitive and always use memory close to memory limits.
+
+   For example, there are some known Java workloads that use 85% of the memory will start to get throttled once this feature is enabled by default. Hence the default 0.8 MemoryThrottlingFactor value may not be a good value.
+
+<br>
+More examples of problematic cases are listed in the table below:
+
+| Limit 1000Mi <br /> Request, factor | current design: memory.high = MemoryThrottlingFactor \* memory.limit (or node allocatable if memory.limit is not set) |
+| -------------------------------- | --------------------------------------------------------------------------------------------------------------------- |
+| request 0Mi, factor 0.6          | 600Mi (early throttling at 600Mi, when 400Mi is unused) |
+| request 500Mi, factor 0.6        | 600Mi (early throttling when memory usage is just 100Mi above requested memory; 400Mi unused) |
+| request 800Mi, factor 0.6        | no throttling (600 < 800 i.e. memory.high < memory.request => no throttling) |
+| request 1Gi, factor 0.6          | max |
+| request 0Mi, factor 0.8          | 800Mi (early throttling at 800Mi, when 200Mi is unused) |
+| request 500Mi, factor 0.8        | 800Mi (early throttling at 800Mi, when 200Mi is unused) |
+| request 850Mi, factor 0.8        | no throttling (800 < 850 i.e. memory.high < memory.request => no throttling) |
+| request 500Gi, factor 0.4        | no throttling (800 < 400  i.e. memory.high < memory.request => no throttling) |
+
+The formula used in Beta implementation eliminates the cases of memory.high being less than memory.request.
+
+###### Quality of Service for Pods
+
+In addition to the change in formula for memory.high, we are also adding the support for memory.high to be set as per `Quality of Service(QoS) for Pod` classes. Following are the different cases:
```

Comment:
Let's explain here why one of the reasons why we decided to add QoS distinction in beta:

maybe something like:
> Based on user feedback in alpha, some users would like to opt-out of MemoryQoS on a per pod basis to ensure there is no early memory throttling. By making user's pods guaranteed, they will be able to do so. Guaranteed are by definition not overcommitted, so `memory.high` does not provide significant value.